File: backend/models/item.go
package models

type Um struct {
	ID   int    `json:"id"`
	Name string `json:"name"`
	Code string `json:"code"`
}

type Vat struct {
	ID                  int     `json:"id"`
	Name                string  `json:"name"`
	Percent             float64 `json:"percent"`
	ExemptionReason     *string `json:"exemption_reason,omitempty"`
	ExemptionReasonCode *string `json:"exemption_reason_code,omitempty"`
	IsActive            bool    `json:"is_active"`
}

type ItemCategory struct {
	ID         int    `json:"id"`
	Name       string `json:"name"`
	IsActive   bool   `json:"is_active"`
	GeneratePn bool   `json:"generate_pn"`
}

type Item struct {
	ID       string        `json:"id"`
	Code     *string       `json:"code,omitempty"`
	Name     string        `json:"name"`
	IsActive bool          `json:"is_active"`
	IsStock  bool          `json:"is_stock"`
	Um       Um            `json:"um"`
	Vat      Vat           `json:"vat"`
	Category *ItemCategory `json:"category,omitempty"`
}

File: frontend/lib/graphql/queries/item.dart
const String getItems = r'''
query getItems($input: GetItemsInput!){   
    getItems(input:$input){
        id,
        code,
        name,
        is_active,
        is_stock,
        um{
            id,
            name
        },
        vat{
            id,
            name,
            percent
        },
        category{
            id,
            name,
            is_active,
            generate_pn
            
        }
        
    }
}
''';

const String getUmList = r'''
query getUmList{
    getUmList{
        id,
        name,
        code,
        is_active
    }
}
''';

const String getVatList = r'''
query GetVatList{
    getVatList{
        id,
        name,
        percent,
        is_active
    }
}
''';

const String getItemCategoryList = r'''
query GetItemCategoryList{
     getItemCategoryList{
        id,
        name,
        is_active,
        generate_pn
    }
}
''';
